runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/BeamFnMapTaskExecutorFactory.java

```diff
@@ -409,16 +413,33 @@ public Node typedApply(RegisterRequestNode input) {
             ptransformIdToSideInputIdToPCollectionView =
                 buildPTransformIdToSideInputIdToPCollectionView(input);
 
+        List<BeamFnApi.ProcessBundleDescriptor> descriptorList =
+            input.getRegisterRequest().getProcessBundleDescriptorList();
+
+        Map<String, String> sdkToDfePCollectionName = new HashMap<>();
+        for (BeamFnApi.ProcessBundleDescriptor descriptor : descriptorList) {
+          for (org.apache.beam.model.pipeline.v1.RunnerApi.PTransform transform :
+              descriptor.getTransformsMap().values()) {
+            for (Map.Entry<String, String> entry : transform.getOutputs().entrySet()) {
+              if (pcollectionSystemToNameMapping.containsKey(entry.getKey())) {
+                sdkToDfePCollectionName.put(
```

Comment:
Is there conceptually a reverse map of this used when populating the ProcessBundleDescriptor? It would be clean  to use same map object can be used when populating the ProcessBundleDescriptor as well. Do you think this change is feasible?

Reply:
My understanding is that we generate this on the fly [during construction of task network.](https://github.com/apache/beam/blob/master/runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/graph/MapTaskToNetworkFunctionTest.java#L110)
Unfortunately, I don't see a convenient way to extract this data on network construction step.